### PR TITLE
Basic image segmentation post (with dask-image)

### PR DESCRIPTION
**Pull request by GenevieveBuckley**
Outline for a blogpost on a basic image segmentation pipeline using dask-image. See https://github.com/dask/dask-blog/issues/47#issuecomment-790119114

Content from github.com/GenevieveBuckley/dask-image-talk-2020

### CODE REVIEW

File: _posts/2021-03-00-image-segmentation.md [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+
+
+```python
+# Measure objects in images
+area = ndmeasure.area(images[:3], label_images, index)
+mean_intensity = ndmeasure.mean(images[:3], label_images, index)
+```
+
+![Label image napari screenshot](../images/2021-image-segmentation/napari-label-image.png)
+
```
Comment (GenevieveBuckley) [inline _posts/2021-03-00-image-segmentation.md]: "Eagle eyed readers will notice that not every nucleus is separated perfectly from its neighbours" (& add link to Nick or Juan's napari interactive segmentation tutorial?)

File: _posts/2021-03-00-image-segmentation.md [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+
```
Comment (GenevieveBuckley) [inline _posts/2021-03-00-image-segmentation.md]: Would be good to have a section on piping the output into a Dask dataframe directly

File: _posts/2021-03-00-image-segmentation.md [hunk 3]
```diff
@@ -0,0 +1,345 @@
...
+![Matplotlib graph of dask-image measurement results: ](../images/2021-image-segmentation/dask-image-matplotlib-output.png)
+
+## Custom functions
+
+What if you want to do something that isn't included?
+
+* scikit-image [apply_parallel()](https://scikit-image.org/docs/dev/api/skimage.util.html#skimage.util.apply_parallel)
+* dask [map_overlap](https://docs.dask.org/en/latest/array-overlap.html?highlight=map_overlap#dask.array.map_overlap) / [map_blocks](https://docs.dask.org/en/latest/array-api.html?highlight=map_blocks#dask.array.map_blocks)
```
Comment (GenevieveBuckley) [inline _posts/2021-03-00-image-segmentation.md]: Needs an example for map_overlap/map_blocks

File: _posts/2021-03-00-image-segmentation.md [hunk 4]
```diff
@@ -0,0 +1,345 @@
...
+* dask [delayed](https://docs.dask.org/en/latest/delayed.html)
```
Comment (GenevieveBuckley) [inline _posts/2021-03-00-image-segmentation.md]: Needs an example of using dask.delayed

File: _posts/2021-03-00-image-segmentation.md [hunk 5]
```diff
@@ -0,0 +1,345 @@
...
+
+
+
+![Matplotlib graph of dask-image measurement results: ](../images/2021-image-segmentation/dask-image-matplotlib-output.png)
+
+## Custom functions
+
+What if you want to do something that isn't included?
+
+* scikit-image [apply_parallel()](https://scikit-image.org/docs/dev/api/skimage.util.html#skimage.util.apply_parallel)
```
Comment (GenevieveBuckley) [inline _posts/2021-03-00-image-segmentation.md]: Better explanation for scikit-image apply_parallel

File: _posts/2021-03-00-image-segmentation.md [hunk 6]
```diff
@@ -0,0 +1,345 @@
...
+theme: twitter
+---
+{% include JB/setup %}
+
+## Executive Summary
+
+A basic image segmentation pipeline
+
+[dask-image](http://image.dask.org/en/latest/)
+
```
Comment (GenevieveBuckley) [inline _posts/2021-03-00-image-segmentation.md]: Do these informative posts need a call to action? (Or was that more for the community building ones)

**Comment by GenevieveBuckley** [comment]
This has been up for long enough that people have had a fair chance to leave comments, so I'm going to go ahead and merge it now.

**Comment by GenevieveBuckley** [comment]
I did think about whether this should have been a runnable notebook in the [dask-examples](https://github.com/dask/dask-examples), but I don't think we can stay under the suggested 2GB memory limitation for it.

**Comment by jakirkham** [comment]
Sorry didn't know this was here 😅

Anyways this looks great! Thanks for doing this 😄

**Comment by GenevieveBuckley** [comment]
> Sorry didn't know this was here sweat_smile

That's probably my fault @jakirkham, I'll try to remember to ping you by name on future posts in your area of interest.

> Anyways this looks great! Thanks for doing this smile

Thanks! It's clunky in two areas:
1. At the end of the pipeline, it would be better to put the results into a dask dataframe (one row per nucleus, with measurements of area & intensity). I wasn't sure how to do that well, but I also didn't spend much time trying. Adding this to examples would be useful.
2. No watershed in dask-image, which is an issue even in this small example. I know that's a technically difficult problem (https://github.com/dask/dask-image/pull/99 is stalled)

**Comment by jakirkham** [comment]
Not at all. It's been a pretty busy month anyways

Idk what parts were tricky with converting to Dask Dataframes, but maybe this simple example helps? If not, would be curious to know where the issues are

```python
In [1]: import numpy as np
   ...: import pandas as pd
   ...: import dask
   ...: import dask.array as da
   ...: import dask.dataframe as dd

In [2]: a = da.stack(3 * [da.arange(1_000, chunks=100)])
   ...: a
Out[2]: dask.array<stack, shape=(3, 1000), dtype=int64, chunksize=(1, 100), chunktype=numpy.ndarray>

In [3]: d = a.T.to_dask_dataframe(["a", "b", "c"])
   ...: d
Out[3]: 
Dask DataFrame Structure:
                    a      b      c
npartitions=10                     
0               int64  int64  int64
100               ...    ...    ...
...               ...    ...    ...
900               ...    ...    ...
999               ...    ...    ...
Dask Name: from-dask, 90 tasks
```

Yeah that's true remembering talking to Juan about this after we finished connected components. It's tricky, but may be doable

**Pull request merged**